File: q70-q79/q72/q72.cpp
class Solution {
public:
    int minDistance(string word1, string word2) {
        int len1 = word1.length(), len2 = word2.length();
        vector<vector<int>> dp(len1 + 1, vector<int>(len2 + 1 , 0));
        for (int i = 0; i <= len1; ++i) {
            dp[i][0] = i;
        }
        for (int j = 0; j <= len2; ++j) {
            dp[0][j] = j;
        }
        
        for (int i = 1; i <= len1; ++i) {
            for (int j = 1; j <= len2; ++j) {
                int replace = word1[i - 1] == word2[j - 1] ? 0 : 1;
                dp[i][j] = min(dp[i-1][j-1] + replace, 
                               min(dp[i-1][j], dp[i][j-1]) + 1);
            }
        }
        return dp[len1][len2];
    } 
};

File: Sequence/q1-q99/q70-q79/q72/q72.py
class Solution:
    def minDistance(self, word1: str, word2: str) -> int:
        n = len(word1)
        m = len(word2)
        dp = [[None]*(n+1) for i in range(m+1)]
        for i in range(m+1):
            dp[i][0] = i
        for i in range(n+1):
            dp[0][i] = i

        for i in range(1,m+1):
            for j in range(1,n+1):
                if word1[j-1] == word2[i-1]:
                    dp[i][j] = dp[i-1][j-1]
                else:
                    dp[i][j] = 1+min(dp[i-1][j-1],dp[i][j-1],dp[i-1][j])
        return dp[m][n]
